store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/FederatedOperation.java

```diff
@@ -16,22 +16,430 @@
+            _getOp().graphIdsCSV(graphIds);
+            return _self();
+        }
+
+        public BuilderParent<INPUT, OUTPUT> mergeFunction(final BiFunction mergeFunction) {
+            _getOp().mergeFunction(mergeFunction);
+            return _self();
+        }
+
+        public BuilderParent<INPUT, OUTPUT> skipFailedFederatedExecution(final boolean skipFailedFederatedExecution) {
+            _getOp().skipFailedFederatedExecution(skipFailedFederatedExecution);
+            return _self();
+        }
+
+        @Override
+        public BuilderParent<INPUT, OUTPUT> _self() {
+            return this;
+        }
+
+        @Override
+        public FederatedOperation<INPUT, OUTPUT> _getOp() {
+            return super._getOp();
+        }
+
+        @Override
+        public BuilderParent<INPUT, OUTPUT> option(final String name, final String value) {
+            return super.option(name, value);
+        }
+
+        @Override
+        public BuilderParent<INPUT, OUTPUT> options(final Map<String, String> options) {
+            return super.options(options);
+        }
+
+        @Override
+        public BuilderParent<INPUT, OUTPUT> userRequestingAdminUsage(final boolean adminRequest) {
+            return super.userRequestingAdminUsage(adminRequest);
+        }
+
+        @Override
+        public FederatedOperation<INPUT, OUTPUT> build() {
+            return super.build();
+        }
+    }
+
+    private static final class BuilderIO<INPUT, OUTPUT> extends FederatedOperation.BuilderParent<INPUT, OUTPUT> {
+        private BuilderIO(final InputOutput<INPUT, Object> op) {
+            super(new FederatedOperation<INPUT, OUTPUT>());
+            FederatedOperation<INPUT, OUTPUT> fedOpIO = this._getOp();
+            fedOpIO.payloadOperation(op);
+        }
+    }
+
+    private static final class BuilderI<INPUT> extends FederatedOperation.BuilderParent<INPUT, Void> {
+        private BuilderI(final Input<INPUT> op) {
+            super(new FederatedOperation<INPUT, Void>());
+            FederatedOperation<INPUT, Void> fedOpI = this._getOp();
+            fedOpI.payloadOperation(op);
+        }
+    }
+
+    private static final class BuilderO<OUTPUT> extends FederatedOperation.BuilderParent<Void, OUTPUT> {
+        private BuilderO(final Output op) {
+            super(new FederatedOperation<Void, OUTPUT>());
+            FederatedOperation<Void, OUTPUT> fedOpO = this._getOp();
+            fedOpO.payloadOperation(op);
+        }
+    }
+
+    private static final class BuilderNeitherIO extends FederatedOperation.BuilderParent<Void, Void> {
+        private BuilderNeitherIO(final Operation op) {
+            super(new FederatedOperation<Void, Void>());
+            FederatedOperation<Void, Void> fedOpO = this._getOp();
+            fedOpO.payloadOperation(op);
+        }
+    }
+
+    @Override
+    public void validateRequiredFieldPresent(final ValidationResult result, final Field field) {
```

Comment:
This function is called in `Operation::validate`, and only gets called on fields with the `@Required` annotation, in this case `payloadOperation`.

The comment is saying `mergeFunction` should be allowed when `payloadOperation instanceof Output`, but I don't see why this is relevant to the function.

Perhaps this is awaiting further changes in the staging branch?

Reply:
I believe this was written when everything was marked as required, since only payload was made "required" I think this logic is no longer required, but no tests flagged up the redundancy of it. It seems safe to delete/remove the redundant logic.